### PR TITLE
Improve the LRE README

### PR DESCRIPTION
**Pull request by MarcusSorealheis**
# Description

Improve the Local Remote Execution README as a continuation of the effort to simplify the documentation in the repository. 

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

## How Has This Been Tested?

Through hacking.

## Checklist

- [x] Updated documentation if needed
- [x] `bazel test //...`  passes locally
- [x] PR is contained in a single commit, using `git amend` see some [docs](https://www.atlassian.com/git/tutorials/rewriting-history)

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)

### CODE REVIEW

File: local-remote-execution/README.md [hunk 1]
```diff
@@ -1,45 +1,65 @@
...
-> Local Remote Execution currently only works on `x86_64-linux`.
+> Local Remote Execution has only been tested on `x86_64-linux`.
 
-First, create an OCI image containing with the toolchains:
+1. Enable a [Flake-based](https://nixos.wiki/wiki/Development_environment_with_nix-shell) Nix Shell within the NativeLink root directory.
 
+```shell
+nix develop
 ```
+2. Create an [OCI image](https://opencontainers.org/) containing with the toolchains:
```
Comment (tyr-one) [inline local-remote-execution/README.md]: "containing the toolchains," unless this is a syntax I don't know

File: local-remote-execution/README.md [hunk 2]
```diff
@@ -1,45 +1,65 @@
...
 
+You should see terminal output beginning with:
+`++ git rev-parse --show-toplevel`
+
+
 The `generate-toolchains` command creates an OCI image from a nix `stdenv` and
 generates toolchain configs from it. The resulting [generated C++ toolchains](
 ./generated/cc/BUILD) have all tools pinned to specific derivations in
 `/nix/store/*`. These paths mirror the ones that you fetched when entering the
 flake development environment, i.e. the tools in the container and in you local
```
Comment (tyr-one) [inline local-remote-execution/README.md]: "in your local"

File: local-remote-execution/README.md [hunk 3]
```diff
@@ -1,45 +1,65 @@
 # Local Remote Execution
```
Comment (tyr-one) [inline local-remote-execution/README.md:2]: I don't know if it should say "Trace Machina" in this document specifically, but we should be sure to say our name with some frequency in documentation generally for branding. Perhaps it should be written "Trace Machina NativeLink" the first time it is named in each document/page?

**Review by tyr-one — COMMENTED** [review]
See my comments in-line. Generally looks good. I like the conversational tone.

File: local-remote-execution/README.md (hunk 1 above)
Comment (MarcusSorealheis) [reply local-remote-execution/README.md]: yikes thanks.

File: local-remote-execution/README.md (hunk 2 above)
Comment (MarcusSorealheis) [reply local-remote-execution/README.md]: Good catch!

File: local-remote-execution/README.md (hunk 3 above)
Comment (MarcusSorealheis) [reply local-remote-execution/README.md:2]: This repo is less about Trace Machina and more about the NativeLink project.

**Review by allada — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637#-:-Np5g9I85rtQwxVnCTHE:bnfp4nl)* status: 1 of 1 LGTMs obtained

___
*[`local-remote-execution/README.md` line 6 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5eflA0TVWfvwLUgxy:-Np5eflA0TVWfvwLUgxz:b-j7aakf) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L6)):*
> ```Markdown
> custom toolchain setups that are transparent, fully hermetic and reproducible
> across machines of the same system architecture. When used in conjunction with Nix,
> Local Remote Execution recreates the remote execution environment in your local
> ```

Maybe `recreates an identical tool chain on your local environment to what remote execution will use`?

The reason is because the environment is not technically the same, but the tool chain is.
___
*[`local-remote-execution/README.md` line 10 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5f6Fu7J3hk5egOCJB:-Np5f6Fu7J3hk5egOCJC:bvo966l) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L10)):*
> ```Markdown
> and local builds while reusing the same cache.
> 
> You must clone the repo to use Local Remote Execution. However, the simplest way to use Local Remote Execution today is to [install and start NativeLink with Cargo](https://github.com/TraceMachina/nativelink?tab=readme-ov-file#-installing-with-cargo).
> ```

This is very confusing because above it says `When used in conjuration with Nix`, but here it says use Cargo.
___
*[`local-remote-execution/README.md` line 14 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5fF65C_hr7lhCVjMZ:-Np5fF65C_hr7lhCVjM_:b-jyr2j7) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L14)):*
> ```Markdown
> 
> 
> - Bazel 7.0.0 or later or another build system that implements [Bazel's RBE](https://bazel.build/remote/rbe)
> ```

nit: Definitely don't need bazel 7. By saying this will deter a lot of people from even attempting.
___
*[`local-remote-execution/README.md` line 29 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5fYls0lvsz2qouUk5:-Np5fYls0lvsz2qouUk6:b-ekkcim) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L29)):*
> ```Markdown
> nix develop
> ```
> 2. Create an [OCI image](https://opencontainers.org/) containing the toolchains:
> ```

nit: Suggest `Generate` instead of `Create`. 
___
*[`local-remote-execution/README.md` line 42 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5g2x95WIJOPMEfAp9:-Np5g2x95WIJOPMEfApA:b66l2tj) ([raw file](https://github.com/TraceMachina/nativelink/blob/39c66781284869d284e4e7168a52b387e2e5f2ae/local-remote-execution/README.md#L42)):*
> ```Markdown
>     --config=lre \
>     --remote_cache=grpc://172.20.255.200:50051 \
>     --define=EXECUTOR=remote \
> ```

nit: I suggest leaving this too. I cant remember what it does, but I do remember the official bazel docs suggest to use it.

**Review by aaronmondal — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)* status: 1 of 1 LGTMs obtained

___
*[`local-remote-execution/README.md` line 10 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5cF8OAKbR6l17qUUq:-Np5cF8OAKbR6l17qUUr:b-ivkmho) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L10)):*
> ```Markdown
> and local builds while reusing the same cache.
> 
> You must clone the repo to use Local Remote Execution. However, the simplest way to use Local Remote Execution today is to [install and start NativeLink with Cargo](https://github.com/TraceMachina/nativelink?tab=readme-ov-file#-installing-with-cargo).
> ```

nit: The part where LRE users most likely trip up is if they don't use actual remote execution but just start a remote cache with a config that invokes builds locally when they should run remotely.

Maybe recommending to have a functional version of the K8s deployment as prerequisite seems like a safer option?
___
*[`local-remote-execution/README.md` line 16 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5_J1s0yS-bFQoGZM4:-Np5_J1s0yS-bFQoGZM5:b-zau7sj) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L16)):*
> ```Markdown
> - Bazel 7.0.0 or later or another build system that implements [Bazel's RBE](https://bazel.build/remote/rbe)
> - Rust 1.73.0 or later
> - Nix 2.19.0 or later
> ```

The only installation requirement is the flake-based nix you mention in point `1.` below. The other tools (such as bazel and Rust) are distributed as part of the flake and don't need to be set up manually.
___
*[`local-remote-execution/README.md` line 62 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5VebE5e26OqkHWJlI:-Np5VebF4gKk0tiniouw:ba4w5gg) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L62)):*
> ```Markdown
>     --config=lre \
>     --remote_cache=grpc://172.20.255.200:50051 \
>     --strategy_regexp .*=remote \
> ```

Does this use the right toolchains? (https://github.com/bazelbuild/bazel/issues/7254).

If this works as intended it looks like a cleaner solution, but I'd like to defer the discussion to a dedicated PR/issue so that we can revisit this after #626.

**Review by MarcusSorealheis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)* status: 1 of 1 LGTMs obtained

___
*[`local-remote-execution/README.md` line 6 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5eflA0TVWfvwLUgxy:-Np5sTLICxPZHrA_CLGH:buqsr8c) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L6)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Maybe `recreates an identical tool chain on your local environment to what remote execution will use`?

The reason is because the environment is not technically the same, but the tool chain is.
</blockquote></details>

I took something pretty close to that in the latest update. Hold tight and thanks for the suggestion.
___
*[`local-remote-execution/README.md` line 10 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5f6Fu7J3hk5egOCJB:-Np5sgxC9nHv3qwEEfNV:brakxrs) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L10)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

This is very confusing because above it says `When used in conjuration with Nix`, but here it says use Cargo.
</blockquote></details>

Done. Good call out.
___
*[`local-remote-execution/README.md` line 10 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5cF8OAKbR6l17qUUq:-Np5t5KL4k5EuzMJkyUK:b-fvr7gy) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L10)):*
<details><summary><i>Previously, aaronmondal (Aaron Siddhartha Mondal) wrote…</i></summary><blockquote>

nit: The part where LRE users most likely trip up is if they don't use actual remote execution but just start a remote cache with a config that invokes builds locally when they should run remotely.

Maybe recommending to have a functional version of the K8s deployment as prerequisite seems like a safer option?
</blockquote></details>

I'm not sure I agree with that. It's mentioned at the end as an option but we are already forcing a lot of third party tools on users.
___
*[`local-remote-execution/README.md` line 14 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5fF65C_hr7lhCVjMZ:-Np5tCDyC094p8c3cuRk:bso9aed) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L14)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

nit: Definitely don't need bazel 7. By saying this will deter a lot of people from even attempting.
</blockquote></details>

correct

**Review by tyr-one — APPROVED** [review]
Changes look good to me!

**Comment by MarcusSorealheis** [comment]
@aaronmondal I addressed your comments. I will merge this so that it is not growing stale.

**Review by aaronmondal — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)* status: 1 of 1 LGTMs obtained

*[a discussion](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np6Iv3ICH6Nfv6BQqBB:-Np6Iv3ICH6Nfv6BQqBC:bvr1wvg) (no related file):*
> I like the conversational tone.

@tyr-one it might be interesting to look at https://github.com/TraceMachina/nativelink/pull/585 which adds a configuration for https://github.com/errata-ai/vale to unify documentation style.

Documentation style for technical documentation is a surprisingly complex topic:
https://learn.microsoft.com/en-us/style-guide/welcome/
https://developers.google.com/style

___
___
*[`local-remote-execution/README.md` line 62 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/637#-Np5VebE5e26OqkHWJlI:-Np6HuNg-oPAbZMADJVE:bmpeg31) ([raw file](https://github.com/TraceMachina/nativelink/blob/24928c1d6c411dc8b654c4dcffd2ec12caf33e43/local-remote-execution/README.md#L62)):*
<details><summary><i>Previously, aaronmondal (Aaron Siddhartha Mondal) wrote…</i></summary><blockquote>

Does this use the right toolchains? (https://github.com/bazelbuild/bazel/issues/7254).

If this works as intended it looks like a cleaner solution, but I'd like to defer the discussion to a dedicated PR/issue so that we can revisit this after #626.
</blockquote></details>

Looks like we pass `--define=EXECUTOR=remote` by default in the `lre` config in the `.bazelrc`.

**Review by aaronmondal — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637#-:-Np6KcWG1_rRS9wVoNlN:bnfp4nl)* status: 2 of 1 LGTMs obtained

**Review by adam-singer — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)* status: 2 of 1 LGTMs obtained

**Review by adam-singer — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637#-:-Np6nOgeANd6yP8LTRGs:bg7igkv)* status: 3 of 1 LGTMs obtained

**Review by allada — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)* status: 3 of 1 LGTMs obtained

**Review by tyr-one — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637)* status: 3 of 1 LGTMs obtained

**Review by MarcusSorealheis — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/637#-:-NpAYF5LBWKwHaRoTfgb:bnfp4nl)* status: <span>:shipit:</span> complete! 4 of 1 LGTMs obtained

**Pull request merged**